db/migrate/20180213163900_create_achievement_users.rb

```diff
@@ -0,0 +1,9 @@
+class CreateAchievementUsers < ActiveRecord::Migration[5.0]
+  def change
+    create_table :achievement_users do |t|
+      t.references :achievement, foreign_key: true, index: true, unique: true
+      t.references :user, foreign_key: true, index: true, unique: true
```

Comment:
This means that each user can only get one achievement :)

Reply:
You probably want to add

`t.index [:achievement_id, :user_id], unique: true`

This will prevent a user from getting the same achievement twice, but each user is allowed to get multiple achievements and an achievement can be given to several users :)